### PR TITLE
Badge preview speed and mode updates on speed or mode change

### PR DESCRIPTION
**Pull request by psalmfill**
Fixes #115

Changes: Add Method to update preview and setup onItemSelectedlisteners on Speed and Mode Spinners to call the method and update preview screen

Screenshots for the change:
speed code
method
mode code

### CODE REVIEW

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 1]
```diff
@@ -17,13 +17,7 @@ import android.support.v7.widget.LinearLayoutManager
...
 import android.view.View
 import android.view.inputmethod.InputMethodManager
-import android.widget.Button
-import android.widget.CheckBox
-import android.widget.EditText
-import android.widget.Spinner
-import android.widget.Toast
-import android.widget.ArrayAdapter
-import android.widget.ProgressBar
+import android.widget.*
```
Comment (yashk2000) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Always optimize imports. Never do imports in this way.

**Comment by yashk2000** [comment]
@psalmfill you are required to submit screenshots of how the app looks and works after the change you made. Screenshots of the code changes are not needed, that is visible anyways in the files changed option. Please update your pr with screenshots of the app.

**Comment by adityastic** [comment]
Good job, sir

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 1 above)
Comment (psalmfill) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Noted @yashk2000

Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Fix this

**Comment by psalmfill** [comment]
@adityastic I have done that

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 2]
```diff
@@ -64,6 +62,7 @@ class MessageActivity : AppCompatActivity() {
 
     private val presenter by lazy { MessagePresenter() }
 
+    private var isTextPreview:Boolean = true
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Follow Indentation

Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
    private var isTextPreview: Boolean = true
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 3]
```diff
@@ -133,6 +134,26 @@ class MessageActivity : AppCompatActivity() {
             else
                 Toast.makeText(this, getString(R.string.select_drawable), Toast.LENGTH_LONG).show()
         }
+        speed.onItemSelectedListener =object : AdapterView.OnItemSelectedListener {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
        speed.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 4]
```diff
@@ -133,6 +134,26 @@ class MessageActivity : AppCompatActivity() {
...
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
+            }
+
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                updatePreview()
+            }
+
+        }
+        mode.onItemSelectedListener =object : AdapterView.OnItemSelectedListener {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
        mode.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 5]
```diff
@@ -133,6 +134,26 @@ class MessageActivity : AppCompatActivity() {
...
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 6]
```diff
@@ -133,6 +134,26 @@ class MessageActivity : AppCompatActivity() {
             else
                 Toast.makeText(this, getString(R.string.select_drawable), Toast.LENGTH_LONG).show()
         }
+        speed.onItemSelectedListener =object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 7]
```diff
@@ -133,6 +134,26 @@ class MessageActivity : AppCompatActivity() {
...
+        }
+        mode.onItemSelectedListener =object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
+            }
+
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                updatePreview()
+            }
+
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 2 above)
Comment (psalmfill) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: @adityastic done

**Comment by adityastic** [comment]
@iamareebjamal you can close this, it had been fixed sir

**Pull request closed without merging**